e2e/test/helpers/TestDeviceCallbackHandler.cs

```diff
@@ -52,33 +58,32 @@ public void Dispose()
             _receivedMessageCallbackSemaphore?.Dispose();
         }
 
-        internal async Task SetDeviceReceiveMethodAsync<T>(string methodName, object deviceResponse, T expectedServiceRequestJson)
+        internal async Task SetDeviceReceiveMethodAndRespondAsync<T,H>(H deviceResponsePayload)
```

Comment:
Decouple the "expected" value from the callback so that we can use this method more generically.